vtex-sites/nextjs.store
# fix: Navigation links on mobile not being clickable

## What's the purpose of this pull request?

Noticed clicking on the links inside the navigation sidebar on mobile wasn't working. I figured it could be related to a recent z-index change from https://github.com/vtex-sites/nextjs.store/pull/212. The fix is to set the lower z-index only for desktop so the search dropdown is shown over the navigation bar. There's no need to set a z-index for mobile because the search dropdown and the navigation links are never displayed at the same time.

Before|After
-|-
![Before, clicking the links stopped working](https://user-images.githubusercontent.com/381395/187303403-22afdcd2-a97e-4dc3-ba78-c82be4094986.gif)|![After, clicking the links worked again](https://user-images.githubusercontent.com/381395/187303224-635d8b91-0949-433d-b608-a625187e5851.gif)

The fix from #212 has been kept:
![The fix from #212 has been kept](https://user-images.githubusercontent.com/381395/187303609-e79d510e-0d2e-4603-9d85-514b355db597.gif)

## How to test it?

Visit https://preview-224--nextjs.preview.vtex.app/ on mobile, open the navigation sidebar and click on one of the links. They all should work again.

## References

- https://github.com/vtex-sites/nextjs.store/pull/212
